### PR TITLE
ColdResumeHandler should be nullptr as its not to be used in default

### PR DESCRIPTION
**Pull request by phoad**
When the coldResumeHandler is not null, the handleUnknownStream function uses this object to save a token and some states. 

In default, we don't need this functionality, so it should be assigned to `nullptr` at the constructor's default parameters. Why to pay the cost if we don't want to use it.

For FireAndForget, we don't need to create/register any stream. Don't pay the cost of creating a stream even though we will not use it.

### CODE REVIEW

File: rsocket/statemachine/RSocketStateMachine.cpp [hunk 1]
```diff
@@ -715,7 +715,9 @@ void RSocketStateMachine::handleUnknownStream(
   // TODO: comparing string versions is odd because from version
   // 10.0 the lexicographic comparison doesn't work
   // we should change the version to struct
-  if (frameSerializer_->protocolVersion() > ProtocolVersion{0, 0} &&
+  DCHECK_GT(frameSerializer_->protocolVersion(), ProtocolVersion(0, 0));
+
+  if (frameType != FrameType::REQUEST_FNF &&
```
Comment (somasun) [inline rsocket/statemachine/RSocketStateMachine.cpp]: Even FNF frames have StreamIds and I think we should validate them.

File: rsocket/RSocket.h [hunk 2]
```diff
@@ -24,8 +24,7 @@ class RSocket {
       std::shared_ptr<RSocketConnectionEvents> connectionEvents =
           std::shared_ptr<RSocketConnectionEvents>(),
       std::shared_ptr<ResumeManager> resumeManager = nullptr,
-      std::shared_ptr<ColdResumeHandler> coldResumeHandler =
-          std::shared_ptr<ColdResumeHandler>(),
+      std::shared_ptr<ColdResumeHandler> coldResumeHandler = nullptr,
```
Comment (lehecka) [inline rsocket/RSocket.h]: this change would break the RSocketClient::resume() (RSocketStateMachine::resume) method.

Comment (phoad) [reply rsocket/RSocket.h]: Actually I only need to update the `createClientFromConnection`, for Thrift integration purposes. Let me remove the remaining updates in this file.

**Comment by phoad** [comment]
I have already updated the project which requires this parameter to be nullptr, to pass nullptr for that parameter. So let me close this diff without landing.

**Pull request closed without merging**